network/servicenetwork/servicenetwork_test.go

```diff
@@ -353,3 +362,22 @@ func TestServiceNetwork_SendCascadeMessage2(t *testing.T) {
 	require.True(t, success)
 }
 */
+
+func TestEcdsaMarshalUnmarshal(t *testing.T) {
+	count := 1000
+	data := testutils.RandomRef()
+	for i := 0; i < count; i++ {
+		privateKey, err := foundation.GeneratePrivateKey()
```

Comment:
Do not use `foundation`. Instead you should use `core.KeyProcessor` component.

Reply:
done